### PR TITLE
adding command-line utility

### PR DESCRIPTION
**Pull request by omgold**
As there seem to be more people interested in this (#1), I would offer to donate my CLI utility for this library.

### CODE REVIEW

File: keepassxc_browser/cli.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python3
+
+import sys
+import os
+import re
+import subprocess
+import argparse
+import json
+import tldextract
```
Comment (hrehfeld) [inline keepassxc_browser/cli.py:9]: missing package dep in setup.py. here? https://github.com/hrehfeld/python-keepassxc-browser/blob/610ea634c2832d6c3948a6851adbc59917bf0119/setup.py#L7

Also, can we avoid this dep?

**Comment by hrehfeld** [comment]
Hmm, I have mixed feelings about this. Wouldn't it be a better idea to maintain this separately? The library is specifically NOT a commandline client (loads of options there iirc?), but something that's easy to use in scripts.

Also see the comments.

And sorry for the long wait :(

File: keepassxc_browser/cli.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        if len(candidates) == 0:
+            print("failure: no pass candidates for URL '%s' found!" % arguments.url, file=sys.stderr)
+            sys.exit(1)
+
+    return candidates
+
+
+def main():
+    argument_parser = argparse.ArgumentParser(description="Fetch credentials from a running KeepassXC instance")
+    argument_parser.add_argument('url')
```
Comment (hrehfeld) [inline keepassxc_browser/cli.py:99]: doc?

File: keepassxc_browser/cli.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            except Exception as e:
+                print("failure: error when trying to write keyfile", e, file=sys.stderr)
+                sys.exit(1)
+    except kpp.ProtocolError as e:
+        print("failure: KeePassXC protocol error:", e, file=sys.stderr)
+        sys.exit(1)
+
+    extract_result = tldextract.extract(url)
+
+    # Try to find candidates using targets in the following order: fully-qualified domain name (includes subdomains),
```
Comment (hrehfeld) [inline keepassxc_browser/cli.py:72]: What's the thought process here? You try to do some sensible matching on the url? What usecase are you covering? At least this needs to be documented (see below @argparse) somehow.

Comment (omgold) [reply keepassxc_browser/cli.py:72]: That is in parts from the qutebrowser scripts for old KeePass. The idea is, if no match is found for the given domain, to search 'upward' the tree of subdomains, up to the last before the TLD.

**Comment by omgold** [comment]
> Hmm, I have mixed feelings about this. Wouldn't it be a better idea to maintain this separately? The library is specifically NOT a commandline client (loads of options there iirc?), but something that's easy to use in scripts.
> 
> Also see the comments.
> 
> And sorry for the long wait :(

That view is of certainly reasonable. It's just that when I found the lib, I also noticed the tickets asking for a CLI utility. As I needed one myself, I created this and then thought I could share it.

Actually I don't have an opinion on whether it should be included with the lib or not. It's just an offer, nothing more.

**Pull request closed without merging**